docs/advanced_topics/documents/title_generation_on_upload.md

```diff
@@ -32,82 +30,174 @@ See MDN for more information about [custom JavaScript events](https://developer.
 
 ## Code examples
 
+For each example below, create the specified external JavaScript file in your app’s static directory, such as static/js/, and reference it in the wagtail_hooks.py file.
+
+### Document specific methods:
+
 ### Adding the file extension to the start of the title
 
 ```python
 # wagtail_hooks.py
 from django.utils.safestring import mark_safe
-
 from wagtail import hooks
-
+from django.templatetags.static import static
 
 @hooks.register("insert_global_admin_js")
 def get_global_admin_js():
-    return mark_safe(
-    """
-    <script>
-    window.addEventListener('DOMContentLoaded', function () {
-        document.addEventListener('wagtail:documents-upload', function(event) {
-            var extension = (event.detail.filename.match(/\.([^.]*?)(?=\?|#|$)/) || [''])[1];
-            var newTitle = '(' + extension.toUpperCase() + ') ' + (event.detail.data.title || '');
-            event.detail.data.title = newTitle;
-        });
+    static_url = static('js/title_with_extension.js')
+    return mark_safe(f'<script src="{static_url}"></script>')
+```
+
+Save the following code as static/js/title_with_extension.js
```

Comment:
I understand, however I thought it might be easier for people to understand if it was right in front of them as they copied the code. Will remove if needed.

Reply:
I've removed them all with just the general note at the top